### ISSUE TITLE
ScreenshotView does not work from contextMenu.

### ISSUE THREAD

**Issue by BlackBird-One** [gualtierofrigerio/ScreenshotSwiftUI#3]
Hi, I'm working on an iOS app using your great ScreenshotSwiftUI library. During the implementation I came across a situation where the library is not working.

It happens when calling the library from contextMenu. Instead of a screenshot, I just get a blank image.

Is there a way to get a screenshot from contextMenu? I've tried everything possible but without success. 

To try it, just modify your InspectSwiftUI project slightly. In the ViewWithEnvironment file, add a contextMenu as follows:

        .screenshotView { screenshotMaker in
            self.screenshotMaker = screenshotMaker
        }
        .contextMenu {
            Button {
                if let screenshotMaker = screenshotMaker {
                    screenshotMaker.screenshot()?.saveToDocuments()
                }
            } label: {
                Label("Share", systemImage: "square.and.arrow.up")
            }
        }

Thank you for your help
Michal Kůs

**Comment by gualtierofrigerio** [comment]
I'm going to try as soon as I can, never tested with a contextMenu.
Just to be sure I understand: you expect to have a screenshot of the view + the context menu on top or only the view without the menu?

**Comment by BlackBird-One** [comment]
It should work like the picture. The user brings up a context menu and sees exactly what is going to be shared. A screenshot must be taken at the moment the "Share" button is pressed to keep the content up to date. 

ShareLink cannot be used for sharing because it requires the shared image to exist when the View is initialized. This means that such a screenshot is out of date - it shows the state before the update. Getter cannot be used for an image, ShareLink does not support it.

![Screenshot4](https://user-images.githubusercontent.com/2879508/194768717-196cacc0-25d1-4647-8cea-5ffc4c9c1921.png)

**Comment by gualtierofrigerio** [comment]
I think I finally reproduced it, if I update the view in one of the context menu buttons and take the screenshot, I have the image without the context menu visible (which I think is fine) but with the old content. Sometimes I don't see the button that I long pressed to show the context menu, maybe is due to the fact it gets rendered differently at that point. 
I noticed that taking the screenshot after a small delay with asyncAfter or with Task { } works, I think because the view eventually gets updated with the new values you put in @State etc. and then the screenshot func is called and can take the correct screenshot.

**Comment by BlackBird-One** [comment]
Of course, the screenshot should not contain a context menu. I've sent this to show the behavior of the application.  The screenshot should look like this:

![1665758380 174564](https://user-images.githubusercontent.com/2879508/195874322-364773bc-3226-4902-80a8-f7963668c43b.png)

When I try to get a screenshot in the context menu I always get this:

![1665757342 7753081](https://user-images.githubusercontent.com/2879508/195873583-e57a40d5-88dc-430d-a2f6-00683b49241b.png)

Taking a screenshot with a small delay is a possible way to go, but not perfect. It is easy for animations to be captured in a transient state.

Thanks for the support and I wish you success in finding a solution.
